src/components/ContactImportation/ContactImportationCard.jsx

```diff
@@ -0,0 +1,196 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Icon } from "cozy-ui/react";
+import IconCross from "../../assets/icons/small-cross.svg";
+import IconFileFailImport from "../../assets/icons/file-fail-import.svg";
+import IconFilePartialImport from "../../assets/icons/file-partial-import.svg";
+import IconFileVcf from "../../assets/icons/file-vcf.svg";
+import IconFileWrongFormat from "../../assets/icons/file-wrong-format.svg";
+import Importation from "../../importation";
+import Status from "../../importation/status";
+
+export default class ContactImportationCard extends React.Component {
+  render() {
+    const {
+      importation,
+      progress,
+      onFileSelected,
+      onFileUnselected
+    } = this.props;
+    const { report } = importation || {};
+    const { t } = this.context;
+
+    switch (importation.status) {
```

Comment:
This `switch/case` in the `render()` function is uncommon.
I see an architectural confusion as `ContactImportationCard` is not a component, it is only a switcher to render this or this component. React and preact could use classical JavaScript function for such needs.

The sole responsibility for `ContactImportationCard` is to render some components according to the variable `importation.status`.

Another smell is the size of the function, you should compartmentalize the rendering components tree. One thing I like to do is to use ["thinking in react" how-to](https://reactjs.org/docs/thinking-in-react.html).

Reply:
> React and preact could use classical JavaScript function for such needs.

Ok, I didn't know arrow function components could access the `t()` function, hence the class.